backend/src/zimfarm_backend/common/schemas/fields.py

```diff
@@ -170,7 +170,9 @@ def validate_secret_url(v: ZIMSecretStr | str) -> ZIMSecretStr:
 ZIMFileName = Annotated[
     str,
     AfterValidator(
-        pattern(r"^(.+?_)([a-z\-]{2,3}?_)(.+_|)([\d]{4}-[\d]{2}|\{period\}).zim$")
+        pattern(
+            r"^([a-z0-9\-\.]+_)([a-z\-]{2,3}?_)([a-z0-9\-\.]+_|)([\d]{4}-[\d]{2}|\{period\}).zim$"
```

Comment:
The language portion must be allowed to be longer than 3 chars. Values like `nds-nl` are now allowed. `([a-z\-]{2,3}?_)` should be replaced with something like `([a-z\-]+?_)` (usually 6 is sufficient, but we do have exceptions like `en-simple`).

Please add such a test cases as well.